docs/src/density_interface.md

```diff
@@ -0,0 +1,21 @@
+# Support for DensityInterface
+
+`Distributions` supports [`DensityInterface`](https://github.com/JuliaMath/DensityInterface.jl) for distributions.
+
+For *single* variate values `x`, `DensityInterface.logdensityof(d::Distribution, x)` is equivalent to `logpdf(d, x)` and `DensityInterface.densityof(d::Distribution, x)` is equivalent to `pdf(d, x)`.
+
+You can use
+```julia
+DensityInterface.logdensityof(Distributions.IIDDensity(d), x)
```

Comment:
`Distributions`'s `Product` can only represent products of univariate distributions, that is annoying here.  So a `IIDDensity` seems to be necessary.

Reply:
Yes - also, for a product we'd need to specify the size of it. From what I understand, Turing needs the "auto-scaling" of the product size (down to one), bases on the shape of `x`. Correct, @devmotion ?

Reply:
Yes, we need the iid assumption of `loglikelihood` and do not fix a size of the implicit iid product distribution a priori. `Product` or a generalization of it are not sufficient here. I tried to explain it in more detail in this comment: https://github.com/JuliaStats/Distributions.jl/pull/1416#issuecomment-963707803

Reply:
I guess for clean semantics, one could just think of `IIDDensity` as a non-normalized probability density over the set of all possible sets of samples, right?

Reply:
Yes

Reply:
Ok, I'll mark this resolved, then? Please feel free to reopen if not.